### PR TITLE
Unify HTTP 404 handling

### LINKED ISSUE #740: [ISSUE] databricks_permission can't be destroyed when job doesn't exist anymore
### Terraform Version
v0.15.4
Databricks provider version:; 0.3.6

### Affected Resource(s)
- databricks_permission

### Expected Behavior
We are trying to remove a databricks_permission resources for a job that has been manually deleted via the console. We would expect the provider to simply skip this step and delete the resource cleanly.

### Actual Behavior
We are getting the following error instead:

```
databricks_permissions.job_permission: Destroying... [id=/jobs/<id>]
╷
│ Error: Job <id> does not exist.
```

Our terraform project is completely unusable now because, as far as we can tell, we can't restore/create a job with a specific id.

### Steps to Reproduce
1. Create a databricks_permissions for a job via Terraform
2. Delete the job via the Databricks console
3. Try to alter or remove the databricks_permissions in Terraform

### Important Factoids
This Terraform project is deployed by gitlab-ci/cd, the statefile isn't readily accessible to be altered (it's not good practice anyway).

### PR DESCRIPTION
**Pull request by nfx**
This commit also fixes #740

### CODE REVIEW

File: access/resource_permissions.go [hunk 1]
```diff
@@ -431,6 +432,10 @@ func ResourcePermissions() *schema.Resource {
 		},
 		DeleteContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 			err := NewPermissionsAPI(ctx, m).Delete(d.Id())
+			if common.IsMissing(err) {
+				log.Printf("[INFO] databricks_permissions[id=%s] dependency is arelady removed on backend", d.Id())
```
Comment (alexott) [inline access/resource_permissions.go:436]: arelady -> already

File: access/resource_permissions.go [hunk 2]
```diff
@@ -337,7 +338,7 @@ func ResourcePermissions() *schema.Resource {
 	readContext := func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
 		id := d.Id()
 		objectACL, err := NewPermissionsAPI(ctx, m).Read(id)
-		if aerr, ok := err.(common.APIError); ok && aerr.IsMissing() {
+		if common.IsMissing(err) {
```
Comment (alexott) [inline access/resource_permissions.go:341]: Would it make sense to add log output here as well?

File: common/http.go [hunk 3]
```diff
@@ -60,6 +60,17 @@ func (apiError APIError) Error() string {
 	return apiError.Message
 }
 
+// IsMissing tells if error is about missing resource
+func IsMissing(err error) bool {
+	if err == nil {
+		return false
+	}
+	if e, ok := err.(APIError); ok && e.IsMissing() {
```
Comment (alexott) [inline common/http.go:68]: Simplify it to the 

e, ok := ….
return ok && e.IsMissing()

?

**Review by alexott — APPROVED** [review]
Lgtm overall, small changes would be useful

File: common/http.go (hunk 3 above)
Comment (nfx) [reply common/http.go:68]: Good one

**Pull request merged**